app/services/edge_cache/bust_article.rb

```diff
@@ -0,0 +1,90 @@
+module EdgeCache
+  class BustArticle < Bust
+    def self.call(article)
+      return unless article
+
+      article.purge
+
+      bust(article.path)
+      bust("/#{article.user.username}")
+      bust("#{article.path}/")
+      bust("#{article.path}?i=i")
+      bust("#{article.path}/?i=i")
+      bust("#{article.path}/comments")
+      bust("#{article.path}?preview=#{article.password}")
+      bust("#{article.path}?preview=#{article.password}&i=i")
+      bust("/#{article.organization.slug}") if article.organization.present?
+      bust_home_pages(article)
+      bust_tag_pages(article)
+      bust("/api/articles/#{article.id}")
+
+      return unless article.collection_id
+
+      article.collection&.articles&.find_each do |collection_article|
```

Comment:
**suggestion (non-blocking):** You probably just copy/pasted this when moving the code, but I think this overuses `&.`: if we got past the guard clause above, we already know we have a collection (referenced by the `collection_id`) and we also know that this collection has at least one article (the one starting this method chain).

Reply:
Updated!